### PR TITLE
[e2e] Use dappeteer optionally

### PR DESCRIPTION
**Pull request by geoknee**
- Partially unksips `rps` integration tests.
- Adds dappeteer, updates tests to use it
- can disable dappeteer (and use a faked out class, and previous faked out web3 object) using env var
- dappeteer does not currently work on CI, but does work locally
- #1516 work in progress to address ^

### CODE REVIEW

File: packages/e2e-tests/puppeteer/__tests__/rps.test.ts [hunk 1]
```diff
@@ -18,25 +14,36 @@ const HEADLESS = getEnvBool('HEADLESS');
...
+let metamaskB: Dappeteer;
 let rpsTabA: Page;
 let rpsTabB: Page;
 
 describe('completes game 1 (challenge by A, challenge by B, resign by B) and begins game 2 ', () => {
   beforeAll(async () => {
-    browserA = await setUpBrowser(HEADLESS, 100); // 100ms sloMo avoids some undiagnosed race conditions
-    browserB = await setUpBrowser(HEADLESS, 100); // 100ms sloMo avoids some undiagnosed race conditions. TODO: remove sloMo and address underlying problem
+    const browserPromiseA = setUpBrowser(HEADLESS, 0);
+    const browserPromiseB = setUpBrowser(HEADLESS, 0); // 100ms sloMo avoids some undiagnosed race conditions. TODO: remove sloMo and address underlying problem
```
Comment (geoknee) [inline packages/e2e-tests/puppeteer/__tests__/rps.test.ts:25]:
```suggestion
    const browserPromiseB = setUpBrowser(HEADLESS, 0);
```
SlowMo not in use.

File: packages/e2e-tests/puppeteer/helpers.ts [hunk 2]
```diff
@@ -102,32 +81,108 @@ export async function waitForAndClickButton(
...
+    new Promise<void>(res => network && res());
+  confirmTransaction: (options: dappeteer.TransactionOptions) => Promise<void> = (
+    options: dappeteer.TransactionOptions
+  ) => new Promise<void>(res => options && res());
+  sign: () => Promise<void> = () => new Promise<void>(res => res());
+  approve: () => Promise<void> = () => new Promise<void>(res => res());
+}
+
+export async function setupFakeWeb3(page: Page, ganacheAccountIndex: number): Promise<void> {
+  // TODO condition on USE_DAPPETEER env var
```
Comment (geoknee) [inline packages/e2e-tests/puppeteer/helpers.ts:104]:
```suggestion

```

File: packages/e2e-tests/.env [hunk 3]
```diff
@@ -1 +1,2 @@
 HEADLESS=true
+USE_DAPPETEER=false
```
Comment (geoknee) [inline packages/e2e-tests/.env:2]: New variable to switch dappeteer on / off

**Comment by andrewgordstewart** [comment]
https://20511-209829093-gh.circle-artifacts.com/0/tmp/seed-download.browser.log

web3torrent was not able to connect to firebase

File: packages/e2e-tests/puppeteer/__tests__/rps.test.ts [hunk 4]
```diff
@@ -18,25 +14,36 @@ const HEADLESS = getEnvBool('HEADLESS');
...
 let rpsTabB: Page;
 
 describe('completes game 1 (challenge by A, challenge by B, resign by B) and begins game 2 ', () => {
   beforeAll(async () => {
-    browserA = await setUpBrowser(HEADLESS, 100); // 100ms sloMo avoids some undiagnosed race conditions
-    browserB = await setUpBrowser(HEADLESS, 100); // 100ms sloMo avoids some undiagnosed race conditions. TODO: remove sloMo and address underlying problem
+    const browserPromiseA = setUpBrowser(HEADLESS, 0);
+    const browserPromiseB = setUpBrowser(HEADLESS, 0);
+
+    ({browser: browserA, metamask: metamaskA} = await browserPromiseA);
```
Comment (tomclose) [inline packages/e2e-tests/puppeteer/__tests__/rps.test.ts:27]: Why the brackets?

File: packages/e2e-tests/puppeteer/helpers.ts [hunk 5]
```diff
@@ -102,32 +81,107 @@ export async function waitForAndClickButton(
...
+}
+
+export async function setUpBrowser(
+  headless: boolean,
+  slowMo?: number
+): Promise<{browser: Browser; metamask: dappeteer.Dappeteer}> {
+  let browser: Browser;
+  let metamask: dappeteer.Dappeteer;
+  if (!USE_DAPPETEER) {
+    // TODO condition on USE_DAPPETEER env var
```
Comment (tomclose) [inline packages/e2e-tests/puppeteer/helpers.ts:133]: ✂️ ?

File: packages/e2e-tests/puppeteer/__tests__/rps.test.ts (hunk 4 above)
Comment (geoknee) [reply packages/e2e-tests/puppeteer/__tests__/rps.test.ts:27]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration

**Pull request merged**